lib/fetchers/url.rb

```diff
@@ -120,6 +121,16 @@ def to_s
 
     private
 
+    def parse_uri(target)
+      if target.is_a?(String)
+        URI.parse(target)
+      elsif target[:url]
+        URI.parse(target[:url])
+      end
+    rescue URI::Error
```

Comment:
If we can't parse the URI for some reason, I'd think we should communicate that back to the user somehow.

Similar logic is used in `resolve_from_string` at line 39, but the circumstances are different: `resolve` is speculative, and is supposed to return nil if the fetcher could not handle the input.

`parse_uri` seems to only be called from `initialize`, which at that point should not be speculative - we've committed to servicing what is thought to be a valid URL. If that is not true, we need to abort.

Reply:
Great catch, I'll rework to attempt to a parse regardless of what is passed.